### ISSUE TITLE
CA1801 False positive for events

### ISSUE THREAD

**Issue by patriksvensson** [dotnet/roslyn-analyzers#2338]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

Example: v2.9.1

#### Diagnostic ID

Example: [CA1801](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801-review-unused-parameters?view=vs-2019)

#### Repro steps

I subscribed to an event in XAML (UWP) that generates a event callback in the code-behind file.

```xaml
<controls:NavigationListView x:Name="MyListView"
    ItemsSource="{x:Bind MenuCommands, Mode=OneWay}"
    ItemInvoked="MyListView_ItemInvoked" />
```

```csharp
private void MyListView_ItemInvoked(object sender, ListViewItem e)
{
   // Code here that does not use sender.
}
```

#### Expected behavior

I did not expect to get a warning saying that `sender` is not being used since I cannot remove the parameter without compilation errors.

#### Actual behavior

I get a warning telling me that not all parameters in the event callback is used (`sender`).

**Comment by mavasani** [comment]
@patriksvensson We special case the analyzer to not flag event handler methods that have a special signature:

1. 2 parameters
2. First parameter is of type `object`
3. Second parameter is of type `EventArgs` or is a sub-type of `EventArgs`

It seems that your mentioned signature does not meet the criteria in 3. Did you mean to use [ListViewItemEventArgs](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.webcontrols.listviewitemeventargs?view=netframework-4.8) instead of `ListViewItem` as the second parameter?

**Comment by mavasani** [comment]
Kindly refer to the suggestion 1. in https://github.com/dotnet/roslyn-analyzers/issues/2593#issuecomment-505965193 - you should rename your parameter with one of the allowed discard symbol names to ensure it is not flagged.
